packages/keystone/apolloErrorFormatter.spec.js

```diff
@@ -287,13 +309,740 @@ describe('safeFormatError hide=false', () => {
+                                        {
+                                            'extensions': { ...fields1, 'message': message4 },
+                                            'message': message4,
+                                            'name': 'GQLError',
+                                            'stack': expect.stringMatching(new RegExp(`^GQLError: ${message4}`)),
+                                        },
+                                    ],
+                                    'message': 'InError',
+                                    'name': 'Error',
+                                    'stack': expect.stringMatching(new RegExp('^Error: InError')),
+                                },
+                            ],
+                            'extensions': { ...fields2, message: message3 },
+                            message: message3,
+                            name: 'GQLError',
+                            'stack': expect.stringMatching(new RegExp(`^GQLError: ${message3}`)),
+                        },
+                    },
+                ],
+            },
+        })
+    })
+    test('safeFormatError(GQLError) change user password real case', () => {
+        const internalErrorMessage = '[error] Update User internal error'
+        const fields1 = {
+            'code': 'INTERNAL_ERROR',
+            'type': 'SUB_GQL_ERROR',
+        }
+        const passwordLengthErrorMessage = 'Password length must be between 8 and 128 characters'
+        const fields2 = {
+            'variable': [
+                'data',
+                'password',
+            ],
+            'code': 'BAD_USER_INPUT',
+            'type': 'INVALID_PASSWORD_LENGTH',
+            'message': 'Password length must be between {min} and {max} characters',
+            'messageForUser': 'api.user.INVALID_PASSWORD_LENGTH',
+            'messageInterpolation': {
+                'min': 8,
+                'max': 128,
+            },
+        }
+        const gqlError2 = new GQLError(fields2, {})
+        const wrappedGQLError2 = new GraphQLError(passwordLengthErrorMessage, null, null, null, null, gqlError2, {})
+        const gqlError1 = new GQLError({ ...fields1, message: internalErrorMessage }, {}, [wrappedGQLError2])
+        const wrappedGQLError1 = new GraphQLError(internalErrorMessage, null, null, null, null, gqlError1, {})
+
+        const result = safeFormatError(wrappedGQLError1)
+        expect(result).toEqual({
+            'name': 'GQLError',
+            'message': '[error] Update User internal error',
+            'extensions': {
+                'code': 'BAD_USER_INPUT',
+                'type': 'INVALID_PASSWORD_LENGTH',
+                'messageForUser': 'Password length must be between 8 and 128 characters',
+                'messageForUserTemplate': 'Password length must be between {min} and {max} characters',
+                'messageForUserTemplateKey': 'api.user.INVALID_PASSWORD_LENGTH',
```

Comment:
Should all this template stuff be transferred / shown to users?

Reply:
It is. Tests just check it by translation keys